### PR TITLE
[issue17] add BooksDetail component

### LINKED ISSUE #17: Create the book detail page
**Is your feature request related to a problem? Please describe.**
N/A

**Describe how the solution should behave**
The page should display each book's details. It should be responsive.

**Screenshots of the design/if any**
image

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by RawandDev**
add BooksDetail component with animating on loading and buttons

### CODE REVIEW

**Review by Niyaz-Dev12 — COMMENTED** [review]
just a note about the UI, if you can add the hover animation to the genres and the 3 extra images it will look better.  LGTM

**Comment by RawandDev** [comment]
> just a note about the UI, if you can add the hover animation to the genres and the 3 extra images it will look better. LGTM

Thanks for the feedback. I added the suggestions and you can now zoom the images. But the edit button is not on the edge because tailwind doesn't have that range to be on the edge of the image.

**Comment by MuhannedNoman** [comment]
image
The Ipad view have extra space under the details? 
image
Also, can you make this icon smaller it looks odd?

**Comment by RawandDev** [comment]
> The Ipad view have extra space under the details?

It is because I set the height to 100vh/max. I will fix this asap.

> Also, can you make this icon smaller it looks odd?

Sure. I will do that.

**Comment by MuhannedNoman** [comment]
Is this caused because of the footer or the component? 
image
Also, the icon still looks the same?

**Comment by RawandDev** [comment]
The footer itself is big if mean the size. I can try to apply a fix on the opened PR related to the footer.

I reduced the edit icon size slightly because if I make it smaller it seems so small and hard to touch on small screens.

**Comment by Niyaz-Dev12** [comment]
> The footer itself is big if mean the size. I can try to apply a fix on the opened PR related to the footer.
> 
> I reduced the edit icon size slightly because if I make it smaller it seems so small and hard to touch on small screens.

i think the issue is the body it's not big enough at ipad screen sizes to cover all the screen that's why the footer doesn't go till the end

File: src/pages/BooksDetail.jsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import { motion } from "framer-motion";
+import AnimateButton from "../components/AnimateButton";
+import coverImg from "../assets/cover.jpg";
+import page1 from "../assets/page1.png";
+import page2 from "../assets/page2.png";
+import page3 from "../assets/page3.png";
+import Zoom from "react-medium-image-zoom";
+import "react-medium-image-zoom/dist/styles.css";
+
+function BooksDetail() {
```
Comment (MuhannedNoman) [inline src/pages/BooksDetail.jsx:12]: How are we getting the data of the book being displayed? Are we passing Id or getting id from the router? Can you handle that here! 
If router (preferable) then just use the route and get the id, and use show how will you fetch the data.

File: package.json [hunk 2]
```diff
@@ -13,6 +13,7 @@
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "react-icons": "^4.3.1",
+    "react-medium-image-zoom": "^4.3.5",
```
Comment (MuhannedNoman) [inline package.json:22]: What are we using this for?

**Review by MuhannedNoman — APPROVED** [review]
Hey @RawandDev good work here, left you some comments. Let me know what you think!

File: src/pages/BooksDetail.jsx (hunk 1 above)
Comment (RawandDev) [reply src/pages/BooksDetail.jsx:12]: I'm currently working on routing and displaying books detail with fake data. Once I'm done I will push and see how it goes.

File: package.json (hunk 2 above)
Comment (RawandDev) [reply package.json:22]: When you click on an image it will zoom the image. I made it with this package.

File: src/pages/BooksDetail.jsx (hunk 1 above)
Comment (RawandDev) [reply src/pages/BooksDetail.jsx:12]: PS: I made changes to the routing and displaying books in this [PR](https://github.com/Iraq-RBC-Capstones-2021/capstone-react-libvery/pull/28).

**Review by MuhannedNoman — DISMISSED** [review]
Make sure to remove the comments and what you wrote before merge. LGTM

**Review by MuhannedNoman — APPROVED** [review]

**Pull request merged**